go/inst/instance_dao.go

```diff
@@ -2492,9 +2496,14 @@ var forceFlushInstanceWriteBuffer = make(chan bool)
 
 func enqueueInstanceWrite(instance *Instance, instanceWasActuallyFound bool, lastError error) {
 	if len(instanceWriteBuffer) == config.Config.InstanceWriteBufferSize {
-		// Signal the "flushing" gorouting that there's work.
+		// Signal the "flushing" goroutine that there's work.
 		// We prefer doing all bulk flushes from one goroutine.
-		forceFlushInstanceWriteBuffer <- true
+		// Non blocking send to avoid blocking goroutines on sending a flush,
+		// if the "flushing" goroutine is not able read is because a flushing is ongoing.
+		select {
+		case forceFlushInstanceWriteBuffer <- true:
+		default:
+		}
```

Comment:
the `select` clause is confusing to read for me. Would you mind rewriting as a goroutine?
```
go func() {
  forceFlushInstanceWriteBuffer <- true
}()
```

Reply:
the reason I use a non-blocking send (using a `select`), it's because I do not want to wait on sending the flush signal if the `forceFlushInstanceWriteBuffer` channel is blocked, since the channel being block means that the flushing process it's already happening, so I don't want to have multiple go-routines "waiting" to flush again a soon as the current flushing finish (the channel is unblocked).